sympy/tensor/array/expressions/arrayexpr_derivatives.py

```diff
@@ -23,7 +23,7 @@ def array_derive(expr, x):
 
 @array_derive.register(Expr) # type: ignore
 def _(expr: Expr, x: Expr):
-    return ZeroArray(*x.shape)
+    return ZeroArray(*x.shape)  # type: ignore
```

Comment:
I would rather fix these things than add `ignore`. This is clearly invalid because an arbitrary Expr does not have a `.shape` attribute.

Reply:
This must be a change to mypy. It couldn't be a PR that was added or else that PR wouldn't have passed tests. And this happening on all PRs. So we either patch it until it gets fixed or wait until someone can fix it and stop all other PRs. What is your preference?

Reply:
The problem was initiated by a change in mypy. I thought that was now addressed (in the ignore sense) but the diff here shows this as a new change in SymPy.

Reply:
#22692 is passing and has the following difference to master (and did not add those differences, to it was a change to master that did this):
```diff
diff --git a/.github/workflows/runtests.yml b/.github/workflows/runtests.yml
index 132075145e..f9c6215388 100644
--- a/.github/workflows/runtests.yml
+++ b/.github/workflows/runtests.yml
@@ -25,7 +25,7 @@ jobs:
         with:
           python-version: '3.9'
       - run: python -m pip install --upgrade pip
-      - run: pip install mpmath flake8
+      - run: pip install mpmath flake8 mypy
 
       - name: Basic code quality tests
         run: bin/test quality
@@ -33,6 +33,9 @@ jobs:
       - name: Run flake8 on the sympy package
         run: flake8 sympy
 
+      - name: Run mypy on the sympy package
+        run: mypy sympy
+
       - name: Detect invalid escapes like '\e'
         run: python -We:invalid -We::SyntaxWarning -m compileall -f -q sympy/
 
@@ -49,28 +52,6 @@ jobs:
       #- name: Make sure all commits have an associated author
       #  run: bin/authors_update.py
 
-
-  # ----------------------------- mypy ----------------------------- #
-
-  mypy:
-    needs: code-quality
-
-    runs-on: ubuntu-20.04
-    steps:
-      - uses: actions/checkout@v2
-        with:
-          # Clone full git history (needed for AUTHORS/mailmap)
-          fetch-depth: '0'
-
-      - uses: actions/setup-python@v2
-        with:
-          python-version: '3.9'
-      - run: python -m pip install --upgrade pip
-      - run: pip install mpmath mypy
-
-      - name: Run mypy on the sympy package
-        run: mypy sympy
-
```

Reply:
@asmeurer , see previous. I think the additions of the red and removal of green diff lines is what made master start failing mypy.